activities/app_configs.py

```diff
@@ -0,0 +1,42 @@
+from django.contrib.auth import get_user_model
+from django.core.urlresolvers import reverse_lazy
+
+from activities.user_data import UserSocialAuthUserData
+from common.app_configs import BaseConnectionAppConfig
+
+
+class UserSocialAuthAppConfig(BaseConnectionAppConfig):
+    """
+    An AppConfig used for activities that connect via UserSocialAuth.
+    """
+
+    def get_user_data(self):
+        outer_self = self
+
+        @property
+        def get_user_data_inner(self):
```

Comment:
@madprime Sorry, this should have been documented better :)

The `UserSocialAuthAppConfig`'s `ready` method adds an attribute to the User model with the name set to the app's `label`.

So in the case of RunKeeper, User gets an attribute called `runkeeper`. The wrapping is to provide that attribute the correct `User` when it's instantiated.

This lets us maintain the old pattern of accessing this data (and provides parity with the way study data is accessed from the `User` object) but makes it so we don't actually have to create a model for activities that use `UserSocialAuth` for their connection (because the model would not contain any additional fields, so it would just be database clutter).